### PR TITLE
Makes Projectiles not hit infrared

### LINKED ISSUE #82052: Physical ammo cannot pass infrared beams
Reporting client version: 515.1633

## Issue Summary
Physical bullets (such as plasma shots and regular bullets/non energy from what i saw) cannot pass infrared beams

## Round ID:
[225660](https://scrubby.melonmesa.com/round/225660)

## Reproduction:

Turn on an invisible infrared beam and shoot in its path, the first shot may go through but the next ones will not pass it

### PR DESCRIPTION
**Pull request by Bilbo367**
## About The Pull Request

Any projectile like bullets cannot hit infrared beams.

## Why It's Good For The Game

Closes https://github.com/tgstation/tgstation/issues/82052

## Changelog

:cl:
fix: projectiles can no longer hit a infrared laser.
/:cl:

### CODE REVIEW

File: code/__DEFINES/_flags.dm [hunk 1]
```diff
@@ -163,6 +163,8 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
 #define LETPASSCLICKS (1<<13)
 /// Allows you to pass over windows and window-adjacent stuff, like windows and windoors. Does not include airlocks with glass in them.
 #define PASSWINDOW (1<<14)
+/// Let projectiles not hit us. **ONLY MEANINGFUL ON pass_flags_self!**
+#define PASSPROJECTILE (1<<15)
```
Comment (Kapu1178) [inline code/__DEFINES/_flags.dm:167]: self-only pass flags are `LETPASS` by convention

**Comment by optimumtact** [comment]
There must be some deeper issue if as the issue reports, the first shot does pass through, but the rest do not, where you able to replicate that behaviour?

**Comment by Bilbo367** [comment]
> There must be some deeper issue if as the issue reports, the first shot does pass through, but the rest do not, where you able to replicate that behavior?

Yes I did find it very finicky, but it mostly depended on the angle most of the time it went through. If you shoot from a almost parallel angle to the beam its much more likely to get hit. Shooting directly perpendicular rarely hits.

**Comment by optimumtact** [comment]
I don't believe this is the right solution here.

**Comment by Bilbo367** [comment]
> I don't believe this is the right solution here.

I'm not sure what else would fix this without changing more root behavior.

**Comment by MrMelbert** [comment]
I also don't believe this is the right solution

**Comment by Bilbo367** [comment]
> > I don't see anything else called beam. What are these beam datums?
> 
> /datum/beam

ahh sorry my vsc was searching was mucked.

**Comment by MrMelbert** [comment]
For the record like, you don't have to go through and do this, I was planning on doing it myself

**Comment by Bilbo367** [comment]
> For the record like, you don't have to go through and do this, I was planning on doing it myself

**Pull request closed without merging**